frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeTemplateSection.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption, StackItem, TextInput } from '@patternfly/react-core';
+import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import { CreatingServingRuntimeObject } from '~/pages/modelServing/screens/types';
+import { TemplateKind } from '~/k8sTypes';
+import {
+  getServingRuntimeDisplayNameFromTemplate,
+  getServingRuntimeNameFromTemplate,
+} from '~/pages/modelServing/customServingRuntimes/utils';
+
+type ServingRuntimeTemplateSectionProps = {
+  data: CreatingServingRuntimeObject;
+  setData: UpdateObjectAtPropAndValue<CreatingServingRuntimeObject>;
+  templates?: TemplateKind[];
+  isEditing?: boolean;
+};
+
+const ServingRuntimeTemplateSection: React.FC<ServingRuntimeTemplateSectionProps> = ({
+  data,
+  setData,
+  templates,
+  isEditing,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+  if (!templates) {
+    return null;
+  }
+
+  const options = templates.map((template) => (
+    <SelectOption
+      key={getServingRuntimeNameFromTemplate(template)}
+      value={getServingRuntimeNameFromTemplate(template)}
+    >
+      {getServingRuntimeDisplayNameFromTemplate(template)}
+    </SelectOption>
+  ));
+
+  const templateSelected =
+    isEditing && !data.servingRuntimeTemplateName
+      ? 'Default runtime'
```

Comment:
Fallback for models deployed with the old method.

Reply:
@vconzola this might be a question, what happens if we edit an old serving runtime?

Reply:
Yeah sure, i wrote this on the fly while reviewing the PR with gage.

I'm talking about the scenario of editing a Serving Runtime deployed without a template:
<img width="814" alt="Screenshot 2023-05-05 at 18 18 16" src="https://user-images.githubusercontent.com/16117276/236512832-48259e19-66b8-4e0b-accc-f14b8c570a38.png">

We don't have the serving runtime name there cause wasn't added in the annotation. This is an example of editing a serving runtime deployed with templates:
<img width="820" alt="Screenshot 2023-05-05 at 18 19 00" src="https://user-images.githubusercontent.com/16117276/236512999-590cd4df-75b8-43dd-b2d1-4190922f7590.png">

My question here is, I added this "Default runtime" label, is that ok, should we use something else?

Reply:
@lucferbux If there actually is a default runtime (one we prefer they select or expect will be used by the majority of users) we should pre-fill the field with the runtime name. If there is not a default runtime the pre-filled text should say, "Select".

Reply:
[UPDATE]: Ok, so it should say "Select cause there's no runtime"

Reply:
I don't know what you mean. The runtime selections in this dropdown should match the list of enabled runtimes in the Serving runtimes table. What is "namespace + model-server"?

Reply:
No sorry, I was confused, I need to stop working in a few minutes 😂🤦‍♂️.

In this specific scenario, it will display "Select" and be disabled, since no template was selected, is it ok?

Reply:
For reference, this is what will display:
<img width="819" alt="Screenshot 2023-05-05 at 19 11 58" src="https://user-images.githubusercontent.com/16117276/236522890-6de07652-ed5f-4ada-8da1-4fa90c93a971.png">

Reply:
I also don't understand this whole "template" concept. I keep seeing it, but no one has ever explained to me the difference between a runtime and a template. Every runtime should have a name and appear in the list of serving runtimes. Period. No exceptions.

Reply:
Ok, the main difference here is the following, templates are this objects:

<img width="1710" alt="Screenshot 2023-05-05 at 19 14 18" src="https://user-images.githubusercontent.com/16117276/236523347-935725a9-2434-47d6-a542-ec2b8dd43f37.png">

The dropdown will display whatever runtime is enabled at that point. Templates are blueprints to create serving runtimes

Serving runtimes are this object:
<img width="1194" alt="Screenshot 2023-05-05 at 19 15 18" src="https://user-images.githubusercontent.com/16117276/236523533-4962147f-1e87-42a2-8f0a-9b33c83a07e4.png">

Are templates instantiated.

The main issue here, is that there will be Serving Runtimes (2) already deployed not based in any template (1), so if we edit them, there's no template (Custom Serving Runtime) in which they are based on, that's why is not selected.

Is that ok?

Reply:
OK. I think the problem we're having communicating is one of terminology. I believe what you're calling a template (1) I'm calling a serving runtime, and what you're calling a serving runtime (2) I'm calling a model server. So, I think the scenario you're describing is what happens if a user tries to edit a model server (2) and the runtime (1) selected when the server was created has been deleted or disabled. Since you're populating the runtime dropdown in the model server modal from the list of enabled runtimes (1), there will be nothing to display for the model server's current runtime. Is that correct? If so, then when editing can we populate the serving runtime field with the name taken from the model server (2) object and then populate the drop down from the enabled serving runtimes (1) list? If a user chose a new runtime and then wanted to go back to the current one, they'd be screwed. But I don't think that's likely to happen very often. And if it did they could always cancel out of the edit modal and start over.

Maybe we can meet and talk thru this on Monday?

Reply:
Yeah, it might be more useful if we meet on Monday to talk through this. But yes, might be the we were having mixed concepts here.